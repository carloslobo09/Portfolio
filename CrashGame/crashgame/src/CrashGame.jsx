import React, { useState } from 'react';
import styled from 'styled-components';
import { Button } from 'react-bootstrap';
import { Howl } from 'howler';
import crashSound from './sounds/crash.wav';

import CrashBoard from './CrashBoard';
import CrashControls from './CrashControls';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 100px;
`;

const crash = new Howl({
    src: [crashSound],
});

function CrashGame() {
    const [gameState, setGameState] = useState({
        multiplier: 1,
        cashedOut: false,
        betAmount: 0,
        roundInProgress: false,
        roundResult: null,
    });

    const handleBetAmountChange = (event) => {
        setGameState((prevState) => ({
            ...prevState,
            betAmount: parseFloat(event.target.value),
        }));
    };

    const handleMultiplierChange = (multiplier) => {
        setGameState((prevState) => ({
            ...prevState,
            multiplier,
        }));
    };

    const handleCashOut = () => {
        setGameState((prevState) => ({
            ...prevState,
            cashedOut: true,
        }));
    };

    const handleRoundStart = () => {
        setGameState((prevState) => ({
            ...prevState,
            roundInProgress: true,
        }))
    }
}