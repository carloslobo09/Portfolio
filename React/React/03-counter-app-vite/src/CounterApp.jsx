import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ({valor}) => {
    const [ counter, setCounter ] = useState(valor)
    
    const HandleAdd = () => {
        setCounter(counter + 1);
    }

    const HandleRest = () => {
        setCounter(counter - 1);
    }

    const HandleReset = () => {
        setCounter(valor);
    }
    
    return (
        <>
            <h1>CounterApp</h1>
            <h2 data-testid='contador'> { counter } </h2>
            <button onClick={ HandleAdd }>+1</button>
            <button onClick={ HandleRest }>-1</button>
            <button aria-label='btn-reset' onClick={ HandleReset }>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    valor : PropTypes.number,
}