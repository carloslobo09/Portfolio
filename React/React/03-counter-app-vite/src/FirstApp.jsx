// import { Fragment } from "react"
import PropTypes from 'prop-types';

const getNameForThis = (valor) => {
    return "Estoy testeando"+valor;
}

export const FirstApp = ({title,subtitle, name}) =>{
    return (
        <>
            <h1 data-testid="title">{ title }</h1>
            <p>{ subtitle }</p>
            <p>{ subtitle }</p>
            <p>{ name }</p>
            <p>{ getNameForThis(" y que miras ura") }</p>
        </>
    )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    title : "No hay titulojiji",
    subtitle: "Aqui tampocosssssssssssssss",
    name: "Ignacio"
}