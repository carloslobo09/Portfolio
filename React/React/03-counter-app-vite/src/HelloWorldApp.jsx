
export const HelloWorldApp = () => {
    return (
        <h1>HOLAAAA</h1>
    );
}
