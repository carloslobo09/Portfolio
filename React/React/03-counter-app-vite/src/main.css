body {
    font-size: 1.3rem;
    font-family: Helvetica, sans-serif;
    background-color: #21232a;
    color: white;
}
button{
    padding: 15px;
    font-size: 1.3rem;
    background-color: gray;
}