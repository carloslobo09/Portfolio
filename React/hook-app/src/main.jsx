import ReactDOM from 'react-dom/client'
import { Layout } from './05-useLayoutEffect/Layout'
import './index.css'
// import { FormWithCustomHook } from './useEffect/FormWithCustomHook'
// import {HooksApp} from './HooksApp.jsx'
// import { CounterWithCustomHook } from './useState/CounterWithCustomHook.jsx'
// import { SimpleForm } from './useEffect/SimpleForm.jsx'

ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
    <Layout/>
  // </React.StrictMode>
)
