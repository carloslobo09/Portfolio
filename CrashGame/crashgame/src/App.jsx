import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import CrashGame from './CrashGame';

function App() {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={CrashGame} />
      </Switch>
    </Router>
  );
}

export default App;