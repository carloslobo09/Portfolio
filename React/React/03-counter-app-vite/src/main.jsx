import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
// import { HelloWorldApp } from './HelloWorldApp'
import { FirstApp } from './FirstApp';
import './main.css';

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        <CounterApp valor={20} />
        {/* <FirstApp title="Hola soy vegeta"/> */}
    </React.StrictMode>
)