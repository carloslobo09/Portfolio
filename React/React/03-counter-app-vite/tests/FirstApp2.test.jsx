import { render,screen } from '@testing-library/react';
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en FirsApp", () => {
    const title = 'Hola soy gey';
    const subtitulo = "Que pasa chavales todo bien todo correcto"

    test("debe hace match con el snapshot", () => {
        const {container} = render(
            <FirstApp title={title} />
        )
        expect(container).toMatchSnapshot();
    });
    
    test("Debe mostrar hola soy gey", () => {
        render(
            <FirstApp title={title} />
        )
        expect(screen.getByText(title)).toBeTruthy();
        // screen.debug(); para ver el objeto
    });
    
    test("Debe mostrar el h1", () => {
        render(
            <FirstApp title={title} />
        )
        expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title)
    });
    
    test("Debe mostrar el subtitulo enviado por props", () => {
        render(
            <FirstApp title={title} subtitle={subtitulo}/>
        )
        expect(screen.getAllByText(subtitulo).length).toBe(2)
    });
    
});
