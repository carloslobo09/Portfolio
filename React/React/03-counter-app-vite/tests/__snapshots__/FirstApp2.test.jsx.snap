// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirsApp debe hace match con el snapshot 1`] = `
<div>
  <h1
    data-testid="title"
  >
    Hola soy gey
  </h1>
  <p>
    Aqui tampocosssssssssssssss
  </p>
  <p>
    Aqui tampocosssssssssssssss
  </p>
  <p>
    Ignacio
  </p>
  <p>
    Estoy testeando y que miras ura
  </p>
</div>
`;
